python/ray/train/jax/jax_trainer.py

```diff
@@ -0,0 +1,160 @@
+            by the ``preprocessor`` if one is provided.
+        preprocessor: A ``ray.data.Preprocessor`` to preprocess the
+            provided datasets.
+        resume_from_checkpoint: A checkpoint to resume training from.
+    """
+
+    def __init__(
+        self,
+        train_loop_per_worker: Union[Callable[[], None], Callable[[Dict], None]],
+        *,
+        train_loop_config: Optional[Dict] = None,
+        jax_config: Optional[JaxConfig] = None,
+        scaling_config: Optional[ScalingConfig] = None,
+        dataset_config: Optional[Dict[str, DatasetConfig]] = None,
+        run_config: Optional[RunConfig] = None,
+        datasets: Optional[Dict[str, GenDataset]] = None,
+        preprocessor: Optional["Preprocessor"] = None,
+        resume_from_checkpoint: Optional[Checkpoint] = None,
+    ):
+        if not jax_config:
+            jax_config = JaxConfig()
+
+        super(JaxTrainer, self).__init__(
+            train_loop_per_worker=train_loop_per_worker,
+            train_loop_config=train_loop_config,
+            backend_config=jax_config,
+            scaling_config=scaling_config,
+            dataset_config=dataset_config,
+            run_config=run_config,
+            datasets=datasets,
+            preprocessor=preprocessor,
+            resume_from_checkpoint=resume_from_checkpoint,
+        )
+
+    @classmethod
+    def _validate_scaling_config(cls, scaling_config: ScalingConfig) -> ScalingConfig:
+        """Return scaling config dataclass after validating updated keys."""
+        ensure_only_allowed_dataclass_keys_updated(
+            dataclass=scaling_config,
+            allowed_keys=cls._scaling_config_allowed_keys,
+        )
+
+        use_gpu = scaling_config.use_gpu
+        use_tpu = (
+            scaling_config.resources_per_worker
+            and scaling_config.resources_per_worker.get("TPU", 0) > 0
+        )
+
+        # cpu parallelism is not supported in jax
+        if not (use_gpu or use_tpu):
+            logger.warning(
+                "CPU parallelism is not supported in jax. "
+                "Please use distributed GPU or TPU training instead. "
+                "Currently, the code is still running on CPU, "
+                "but there is no distributed training happening."
+            )
+
+        # change the default value of `placment_strategy` to `SPREAD`.
+        # using `SPREAD`: the jaxtrainer will spread the workers across different nodes.
+        # between the nodes, the connections is to be established by NCCL;
+        # within the nodes, jax can already enable communicates automatically.
+        # so, no end to use jax backend to make the connections.
+        # NOT `STRICT_SPREAD`: if the users use the trainer on one of GPU nodes instead
+        # of on the extra cpu nodes, then `STRICT_SPREAD` will hang on for the resources
+        if "PACK" in scaling_config.placement_strategy:
```

Comment:
i would suggest we just throw exception here rather than overriding values.

1) AIR prints a lot of logs (some logs not surfaced to driver terminal as well) that user might not see this
2) Unlike other warnings that we choose a default value for users, this is changing user behavior implicitly
3) Cost of throwing exception here is quite small since it happens at constructor 

The other example we have in AIR is TensorArray casting, and we throw exception loud and quick when encountered: https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/air/util/tensor_extensions/pandas.py?L720

But runtime behavior and flag is always consistent with user config, rather than us magically changing them.

Reply:
make sense, one thing here is that the placement group default strategy is `PACK`. In the case where the users don't provide anything for the strategy, it would be good to let the code running. I wonder whether there is an alternative way to change the default strategy here to `SPREAD`.

Reply:
@jiaodong definitely agree with all the points you've made. Ideally, we can set the default value to be `SPREAD` and raise an exception if the user explicitly sets anything else.

However, since the current existing pattern is for all Trainers to share the same `ScalingConfig`, it's a pretty bad experience to have the user explicitly specify the `ScalingConfig` with `SPREAD` in order to use this Trainer.

My vote would be to just force convert to `SPREAD`, ignoring whatever value is passed in, until we have a better story around Trainer specific `ScalingConfig`s. Thoughts?

Reply:
If we do that we should at least throw a warning about it

Reply:
make sense